scripts/reset-test-db.sh

```diff
@@ -24,35 +24,55 @@
+
 # Ensure the test database exists.
-echo "CREATE DATABASE IF NOT EXISTS edxtest;" | mysql -u root
+for db in "${database_order[@]}"; do
+    echo "CREATE DATABASE IF NOT EXISTS ${databases[$db]};" | mysql -u root
 
-# Clear out the test database
-#
-# We are using the django-extensions's reset_db command which uses "DROP DATABASE" and
-# "CREATE DATABASE" in case the tests are being run in an environment (e.g. devstack
-# or a jenkins worker environment) that already ran tests on another commit that had
-# different migrations that created, dropped, or altered tables.
-echo "Issuing a reset_db command to the bok_choy MySQL database."
-./manage.py lms --settings bok_choy reset_db --traceback --noinput
+    # Clear out the test database
+    #
+    # We are using the django-extensions's reset_db command which uses "DROP DATABASE" and
+    # "CREATE DATABASE" in case the tests are being run in an environment (e.g. devstack
+    # or a jenkins worker environment) that already ran tests on another commit that had
+    # different migrations that created, dropped, or altered tables.
+    echo "Issuing a reset_db command to the bok_choy MySQL database."
+    ./manage.py lms --settings bok_choy reset_db --traceback --noinput --router $db
+
+    # If there are cached database schemas/data, load them
+    if [[ ! -f $DB_CACHE_DIR/bok_choy_schema_$db.sql || ! -f $DB_CACHE_DIR/bok_choy_data_$db.json || ! -f $DB_CACHE_DIR/bok_choy_migrations_data_$db.sql ]]; then
+        echo "Missing $DB_CACHE_DIR/bok_choy_schema_$db.sql or $DB_CACHE_DIR/bok_choy_data_$db.json, or $DB_CACHE_DIR/bok_choy_migrations_data_$db.sql rebuilding cache"
+        REBUILD_CACHE=true
+    fi
+
+done
 
 # If there are cached database schemas/data, load them
-if [[ -f $DB_CACHE_DIR/bok_choy_schema.sql && -f $DB_CACHE_DIR/bok_choy_migrations_data.sql && -f $DB_CACHE_DIR/bok_choy_data.json ]]; then
+if [[ -z $REBUILD_CACHE ]]; then
 
     echo "Found the bok_choy DB cache files. Loading them into the database..."
-    # Load the schema, then the data (including the migration history)
-    echo "Loading the schema from the filesystem into the MySQL DB."
-    mysql -u root edxtest < $DB_CACHE_DIR/bok_choy_schema.sql
-    echo "Loading the migration data from the filesystem into the MySQL DB."
-    mysql -u root edxtest < $DB_CACHE_DIR/bok_choy_migrations_data.sql
-    echo "Loading the fixture data from the filesystem into the MySQL DB."
-    ./manage.py lms --settings bok_choy loaddata $DB_CACHE_DIR/bok_choy_data.json
-
-    # Re-run migrations to ensure we are up-to-date
-    echo "Running the lms migrations on the bok_choy DB."
-    ./manage.py lms --settings bok_choy migrate --traceback --noinput
-    echo "Running the cms migrations on the bok_choy DB."
-    ./manage.py cms --settings bok_choy migrate --traceback --noinput
+
+    for db in "${database_order[@]}"; do
+        # Load the schema, then the data (including the migration history)
```

Comment:
Would be good to specify which database all the echoes correspond to. For simplicity, you could even just have a start/end echoes so we know which one we're talking about.